core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -147,21 +239,39 @@ public GeoDistanceSortBuilder setNestedPath(String nestedPath) {
         this.nestedPath = nestedPath;
         return this;
     }
+    
+    /**
+     * Returns the nested path if sorting occurs on a field that is inside a nested object. By default when sorting on a
+     * field inside a nested object, the nearest upper nested object is selected as nested path.
+     */
+    public String getNestedPath() {
+        return this.nestedPath;
+    }
 
     public GeoDistanceSortBuilder coerce(boolean coerce) {
         this.coerce = coerce;
         return this;
     }
 
+    public boolean coerce() {
+        return this.coerce;
+    }
+
     public GeoDistanceSortBuilder ignoreMalformed(boolean ignoreMalformed) {
-        this.ignoreMalformed = ignoreMalformed;
+        if (coerce == false) {
+            this.ignoreMalformed = ignoreMalformed;
+        }
         return this;
     }
+    
+    public boolean ignoreMalformed() {
+        return this.ignoreMalformed;
+    }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("_geo_distance");
-        if (geohashes.size() == 0 && points.size() == 0) {
+        builder.startObject(NAME);
+        if (points.size() == 0) {
```

Comment:
This kind of validation feels odd in toXContent. Can we make point/point mandatory constructor args?